### PR TITLE
N°3287 - Set from display name / label in action email

### PR DESCRIPTION
**Pull request by larhip**
With this PR you should be able to set the display name of the email sender (as well as reply to).

Two more attributes (from_label & reply_to_label) added to ActionEmail
dictionaries has been updated (at least with placeholders)

### CODE REVIEW

**Review by Hipska — DISMISSED** [review]

File: dictionaries/nl.dictionary.itop.core.php [hunk 1]
```diff
@@ -513,8 +513,12 @@
 	'Class:ActionEmail/Attribute:test_recipient+' => 'Bestemming als de status op "Test" staat',
 	'Class:ActionEmail/Attribute:from' => 'Van',
 	'Class:ActionEmail/Attribute:from+' => 'Wordt gebruikt in de hoofdtekst van de e-mail (headers)',
+	'Class:ActionEmail/Attribute:from_label' => 'From (label)~~',
+	'Class:ActionEmail/Attribute:from_label+' => 'Sender display name will be sent into the email header~~',
 	'Class:ActionEmail/Attribute:reply_to' => 'Antwoord',
 	'Class:ActionEmail/Attribute:reply_to+' => 'Wordt gebruikt in de hoofdtekst van de e-mail (headers)',
+	'Class:ActionEmail/Attribute:reply_to_label' => 'Reply to (label)~~',
+	'Class:ActionEmail/Attribute:reply_to_label+' => 'Reply to display name will be sent into the email header~~',
```
Comment (Hipska) [inline dictionaries/nl.dictionary.itop.core.php]: Dutch translations:
```suggestion
	'Class:ActionEmail/Attribute:from' => 'Van (e-mail)',
	'Class:ActionEmail/Attribute:from+' => 'Wordt gebruikt in de hoofdtekst van de e-mail (headers)',
	'Class:ActionEmail/Attribute:from_label' => 'Van (label)',
	'Class:ActionEmail/Attribute:from_label+' => 'Wordt gebruikt in de hoofdtekst van de e-mail (headers)',
	'Class:ActionEmail/Attribute:reply_to' => 'Antwoord (e-mail)',
	'Class:ActionEmail/Attribute:reply_to+' => 'Wordt gebruikt in de hoofdtekst van de e-mail (headers)',
	'Class:ActionEmail/Attribute:reply_to_label' => 'Antwoord (label)',
	'Class:ActionEmail/Attribute:reply_to_label+' => 'Wordt gebruikt in de hoofdtekst van de e-mail (headers)',
```

**Comment by Hipska** [comment]
I think you should mark the translations of `Class:ActionEmail/Attribute:from` and `Class:ActionEmail/Attribute:reply_to` (and the descriptions as well) as being new or to be translated (`~~`)..

**Review by Hipska — DISMISSED** [review]

**Comment by larhip** [comment]
Thanks @Hipska for dutch translations and the hint regarding the existing translations.
I updated the dict files and squash all changes into one commit

**Comment by Molkobain** [comment]
Looks all good, thanks @larhip !

File: dictionaries/es_cr.dictionary.itop.core.php [hunk 2]
```diff
@@ -507,8 +507,12 @@
 	'Class:ActionEmail/Attribute:test_recipient+' => 'Destinatario en caso que el Estatus sea "En pruebas"',
 	'Class:ActionEmail/Attribute:from' => 'Remitente',
 	'Class:ActionEmail/Attribute:from+' => 'Será enviando en el encabezado del Correo Electrónico',
+	'Class:ActionEmail/Attribute:from_label' => 'From (label)~~',
+	'Class:ActionEmail/Attribute:from_label+' => 'Sender display name will be sent into the email header~~',
 	'Class:ActionEmail/Attribute:reply_to' => 'Responder a',
 	'Class:ActionEmail/Attribute:reply_to+' => 'Será enviando en el encabezado del Correo Electrónico',
```
Comment (Hipska) [inline dictionaries/es_cr.dictionary.itop.core.php]:
```suggestion
	'Class:ActionEmail/Attribute:from' => 'Remitente~~',
	'Class:ActionEmail/Attribute:from+' => 'Será enviando en el encabezado del Correo Electrónico~~',
	'Class:ActionEmail/Attribute:from_label' => 'From (label)~~',
	'Class:ActionEmail/Attribute:from_label+' => 'Sender display name will be sent into the email header~~',
	'Class:ActionEmail/Attribute:reply_to' => 'Responder a~~',
	'Class:ActionEmail/Attribute:reply_to+' => 'Será enviando en el encabezado del Correo Electrónico~~',
```

**Review by Hipska — CHANGES_REQUESTED** [review]

File: dictionaries/es_cr.dictionary.itop.core.php (hunk 2 above)
Comment (larhip) [reply dictionaries/es_cr.dictionary.itop.core.php]: oh.. I missed them thanks for that!

**Comment by Molkobain** [comment]
Note for myself: Check derivated classes from ActionEmail to see if the new attributes should be added in the details zlist (eg. approval modules)

**Comment by Hipska** [comment]
I was wondering, couldn't the FROM name come from the logged in user? (Similar like here on GitHub notifications)

**Comment by larhip** [comment]
Not sure whether this is always intended behavior..
Anyhow if you want to do it in this way you should be able to configure it with this PR (by setting from_label to a placeholder based on current user i.e. 
`$current_contact->friendlyname$`)

**Review by Hipska — APPROVED** [review]
Aha, that's perfect then!

**Review by Molkobain — APPROVED** [review]

File: dictionaries/cs.dictionary.itop.core.php [hunk 3]
```diff
@@ -507,10 +507,14 @@
...
-	'Class:ActionEmail/Attribute:from' => 'Odesílatel',
-	'Class:ActionEmail/Attribute:from+' => '',
-	'Class:ActionEmail/Attribute:reply_to' => 'Odpověď na',
-	'Class:ActionEmail/Attribute:reply_to+' => '',
+	'Class:ActionEmail/Attribute:from' => 'Odesílatel~~',
+	'Class:ActionEmail/Attribute:from+' => '~~',
+	'Class:ActionEmail/Attribute:from_label' => 'From (label)~~',
+	'Class:ActionEmail/Attribute:from_label+' => 'Sender display name will be sent into the email header~~',
+	'Class:ActionEmail/Attribute:reply_to' => 'Odpověď na~~',
+	'Class:ActionEmail/Attribute:reply_to+' => '~~',
```
Comment (Molkobain) [inline dictionaries/cs.dictionary.itop.core.php:515]: Why did you change the translations and added "~~" at the end?

File: dictionaries/da.dictionary.itop.core.php [hunk 4]
```diff
@@ -505,10 +505,14 @@
...
-	'Class:ActionEmail/Attribute:from' => 'Fra',
-	'Class:ActionEmail/Attribute:from+' => 'Afsender af emailen',
-	'Class:ActionEmail/Attribute:reply_to' => 'Svar til',
-	'Class:ActionEmail/Attribute:reply_to+' => 'Svar sendes til',
+	'Class:ActionEmail/Attribute:from' => 'Fra~~',
+	'Class:ActionEmail/Attribute:from+' => 'Afsender af emailen~~',
+	'Class:ActionEmail/Attribute:from_label' => 'From (label)~~',
+	'Class:ActionEmail/Attribute:from_label+' => 'Sender display name will be sent into the email header~~',
+	'Class:ActionEmail/Attribute:reply_to' => 'Svar til~~',
+	'Class:ActionEmail/Attribute:reply_to+' => 'Svar sendes til~~',
```
Comment (Molkobain) [inline dictionaries/da.dictionary.itop.core.php:513]: Same comment

File: dictionaries/es_cr.dictionary.itop.core.php [hunk 5]
```diff
@@ -505,10 +505,14 @@
...
-	'Class:ActionEmail/Attribute:from' => 'Remitente',
-	'Class:ActionEmail/Attribute:from+' => 'Será enviando en el encabezado del Correo Electrónico',
-	'Class:ActionEmail/Attribute:reply_to' => 'Responder a',
-	'Class:ActionEmail/Attribute:reply_to+' => 'Será enviando en el encabezado del Correo Electrónico',
+	'Class:ActionEmail/Attribute:from' => 'Remitente~~',
+	'Class:ActionEmail/Attribute:from+' => 'Será enviando en el encabezado del Correo Electrónico~~',
+	'Class:ActionEmail/Attribute:from_label' => 'From (label)~~',
+	'Class:ActionEmail/Attribute:from_label+' => 'Sender display name will be sent into the email header~~',
+	'Class:ActionEmail/Attribute:reply_to' => 'Responder a~~',
+	'Class:ActionEmail/Attribute:reply_to+' => 'Será enviando en el encabezado del Correo Electrónico~~',
```
Comment (Molkobain) [inline dictionaries/es_cr.dictionary.itop.core.php:513]: Same comment

File: dictionaries/fr.dictionary.itop.core.php [hunk 6]
```diff
@@ -503,10 +503,14 @@
...
-	'Class:ActionEmail/Attribute:from' => 'De',
-	'Class:ActionEmail/Attribute:from+' => '',
-	'Class:ActionEmail/Attribute:reply_to' => 'Répondre à',
-	'Class:ActionEmail/Attribute:reply_to+' => '',
+	'Class:ActionEmail/Attribute:from' => 'De~~',
+	'Class:ActionEmail/Attribute:from+' => '~~',
+	'Class:ActionEmail/Attribute:from_label' => 'From (label)~~',
+	'Class:ActionEmail/Attribute:from_label+' => 'Sender display name will be sent into the email header~~',
+	'Class:ActionEmail/Attribute:reply_to' => 'Répondre à~~',
+	'Class:ActionEmail/Attribute:reply_to+' => '~~',
```
Comment (Molkobain) [inline dictionaries/fr.dictionary.itop.core.php:511]: Same comment

File: dictionaries/hu.dictionary.itop.core.php [hunk 7]
```diff
@@ -503,10 +503,14 @@
...
-	'Class:ActionEmail/Attribute:from' => 'Feladó',
-	'Class:ActionEmail/Attribute:from+' => '',
-	'Class:ActionEmail/Attribute:reply_to' => 'Válasz',
-	'Class:ActionEmail/Attribute:reply_to+' => '',
+	'Class:ActionEmail/Attribute:from' => 'Feladó~~',
+	'Class:ActionEmail/Attribute:from+' => '~~',
+	'Class:ActionEmail/Attribute:from_label' => 'From (label)~~',
+	'Class:ActionEmail/Attribute:from_label+' => 'Sender display name will be sent into the email header~~',
+	'Class:ActionEmail/Attribute:reply_to' => 'Válasz~~',
+	'Class:ActionEmail/Attribute:reply_to+' => '~~',
```
Comment (Molkobain) [inline dictionaries/hu.dictionary.itop.core.php:511]: Same comment

File: dictionaries/it.dictionary.itop.core.php [hunk 8]
```diff
@@ -505,10 +505,14 @@
...
-	'Class:ActionEmail/Attribute:from' => 'Da',
-	'Class:ActionEmail/Attribute:from+' => '',
-	'Class:ActionEmail/Attribute:reply_to' => 'Rispondi A',
-	'Class:ActionEmail/Attribute:reply_to+' => '',
+	'Class:ActionEmail/Attribute:from' => 'Da~~',
+	'Class:ActionEmail/Attribute:from+' => '~~',
+	'Class:ActionEmail/Attribute:from_label' => 'From (label)~~',
+	'Class:ActionEmail/Attribute:from_label+' => 'Sender display name will be sent into the email header~~',
+	'Class:ActionEmail/Attribute:reply_to' => 'Rispondi A~~',
+	'Class:ActionEmail/Attribute:reply_to+' => '~~',
```
Comment (Molkobain) [inline dictionaries/it.dictionary.itop.core.php:513]: Same comment

File: dictionaries/ja.dictionary.itop.core.php [hunk 9]
```diff
@@ -503,10 +503,14 @@
...
-	'Class:ActionEmail/Attribute:from' => 'From',
-	'Class:ActionEmail/Attribute:from+' => '電子メールのヘッダーに挿入されます',
-	'Class:ActionEmail/Attribute:reply_to' => 'Reply to',
-	'Class:ActionEmail/Attribute:reply_to+' => '電子メールのヘッダーに挿入されます',
+	'Class:ActionEmail/Attribute:from' => 'From~~',
+	'Class:ActionEmail/Attribute:from+' => '電子メールのヘッダーに挿入されます~~',
+	'Class:ActionEmail/Attribute:from_label' => 'From (label)~~',
+	'Class:ActionEmail/Attribute:from_label+' => 'Sender display name will be sent into the email header~~',
+	'Class:ActionEmail/Attribute:reply_to' => 'Reply to~~',
+	'Class:ActionEmail/Attribute:reply_to+' => '電子メールのヘッダーに挿入されます~~',
```
Comment (Molkobain) [inline dictionaries/ja.dictionary.itop.core.php:511]: Same comment

File: dictionaries/pt_br.dictionary.itop.core.php [hunk 10]
```diff
@@ -505,10 +505,14 @@
...
-	'Class:ActionEmail/Attribute:from' => 'De',
-	'Class:ActionEmail/Attribute:from+' => 'Será enviado para o cabeçalho de email',
-	'Class:ActionEmail/Attribute:reply_to' => 'Responder para',
-	'Class:ActionEmail/Attribute:reply_to+' => 'Será enviado para o cabeçalho de email',
+	'Class:ActionEmail/Attribute:from' => 'De~~',
+	'Class:ActionEmail/Attribute:from+' => 'Será enviado para o cabeçalho de email~~',
+	'Class:ActionEmail/Attribute:from_label' => 'From (label)~~',
+	'Class:ActionEmail/Attribute:from_label+' => 'Sender display name will be sent into the email header~~',
+	'Class:ActionEmail/Attribute:reply_to' => 'Responder para~~',
+	'Class:ActionEmail/Attribute:reply_to+' => 'Será enviado para o cabeçalho de email~~',
```
Comment (Molkobain) [inline dictionaries/pt_br.dictionary.itop.core.php:513]: Same comment

File: dictionaries/ru.dictionary.itop.core.php [hunk 11]
```diff
@@ -492,10 +492,14 @@
...
-	'Class:ActionEmail/Attribute:from' => 'От',
-	'Class:ActionEmail/Attribute:from+' => 'Будет отправлено в заголовке email',
-	'Class:ActionEmail/Attribute:reply_to' => 'Ответить на',
-	'Class:ActionEmail/Attribute:reply_to+' => 'Будет отправлено в заголовке email',
+	'Class:ActionEmail/Attribute:from' => 'От~~',
+	'Class:ActionEmail/Attribute:from+' => 'Будет отправлено в заголовке email~~',
+	'Class:ActionEmail/Attribute:from_label' => 'From (label)~~',
+	'Class:ActionEmail/Attribute:from_label+' => 'Sender display name will be sent into the email header~~',
+	'Class:ActionEmail/Attribute:reply_to' => 'Ответить на~~',
+	'Class:ActionEmail/Attribute:reply_to+' => 'Будет отправлено в заголовке email~~',
```
Comment (Molkobain) [inline dictionaries/ru.dictionary.itop.core.php:500]: Same comment

File: dictionaries/sk.dictionary.itop.core.php [hunk 12]
```diff
@@ -502,10 +502,14 @@
...
-	'Class:ActionEmail/Attribute:from' => 'Od',
-	'Class:ActionEmail/Attribute:from+' => '',
-	'Class:ActionEmail/Attribute:reply_to' => 'Odpoveď na',
-	'Class:ActionEmail/Attribute:reply_to+' => '',
+	'Class:ActionEmail/Attribute:from' => 'Od~~',
+	'Class:ActionEmail/Attribute:from+' => '~~',
+	'Class:ActionEmail/Attribute:from_label' => 'From (label)~~',
+	'Class:ActionEmail/Attribute:from_label+' => 'Sender display name will be sent into the email header~~',
+	'Class:ActionEmail/Attribute:reply_to' => 'Odpoveď na~~',
+	'Class:ActionEmail/Attribute:reply_to+' => '~~',
```
Comment (Molkobain) [inline dictionaries/sk.dictionary.itop.core.php:510]: Same comment

File: dictionaries/tr.dictionary.itop.core.php [hunk 13]
```diff
@@ -513,10 +513,14 @@
...
-	'Class:ActionEmail/Attribute:from' => 'Kimden',
-	'Class:ActionEmail/Attribute:from+' => 'e-posta başlığında gönderilecek',
-	'Class:ActionEmail/Attribute:reply_to' => 'Yanıtla',
-	'Class:ActionEmail/Attribute:reply_to+' => 'e-posta başlığında gönderilecek',
+	'Class:ActionEmail/Attribute:from' => 'Kimden~~',
+	'Class:ActionEmail/Attribute:from+' => 'e-posta başlığında gönderilecek~~',
+	'Class:ActionEmail/Attribute:from_label' => 'From (label)~~',
+	'Class:ActionEmail/Attribute:from_label+' => 'Sender display name will be sent into the email header~~',
+	'Class:ActionEmail/Attribute:reply_to' => 'Yanıtla~~',
+	'Class:ActionEmail/Attribute:reply_to+' => 'e-posta başlığında gönderilecek~~',
```
Comment (Molkobain) [inline dictionaries/tr.dictionary.itop.core.php:521]: Same comment

File: dictionaries/zh_cn.dictionary.itop.core.php [hunk 14]
```diff
@@ -504,10 +504,14 @@
...
 	'Class:ActionEmail/Attribute:test_recipient+' => 'Detination in case status is set to "Test"',
-	'Class:ActionEmail/Attribute:from' => '发件人',
-	'Class:ActionEmail/Attribute:from+' => 'Will be sent into the email header',
-	'Class:ActionEmail/Attribute:reply_to' => '回复到',
-	'Class:ActionEmail/Attribute:reply_to+' => 'Will be sent into the email header',
+	'Class:ActionEmail/Attribute:from' => '发件人~~',
+	'Class:ActionEmail/Attribute:from+' => 'Will be sent into the email header~~',
+	'Class:ActionEmail/Attribute:from_label' => 'From (label)~~',
+	'Class:ActionEmail/Attribute:from_label+' => 'Sender display name will be sent into the email header~~',
+	'Class:ActionEmail/Attribute:reply_to' => '回复到~~',
```
Comment (Molkobain) [inline dictionaries/zh_cn.dictionary.itop.core.php:511]: Same comment

**Review by Molkobain — CHANGES_REQUESTED** [review]

File: dictionaries/cs.dictionary.itop.core.php (hunk 3 above)
Comment (larhip) [reply dictionaries/cs.dictionary.itop.core.php:515]: In the German and English dictonary files I changed the translations from "From" to "From (email)", to make clear, you have to configure the sender mail here (and the sender label in a different attribute).
I took the idea of @Hipska and added "~~" to the existing translations:
> I think you should mark the translations of `Class:ActionEmail/Attribute:from` and `Class:ActionEmail/Attribute:reply_to` (and the descriptions as well) as being new or to be translated (`~~`)..

Comment (Molkobain) [reply dictionaries/cs.dictionary.itop.core.php:515]: Oh, ok, sorry I forgot that. Then it's all good. Thank you for the reminder!

**Review by Molkobain — APPROVED** [review]

**Pull request merged**